js/src/tests/traceable_langchain.test.ts

```diff
@@ -0,0 +1,275 @@
+import { traceable } from "../traceable.js";
+import { getAssumedTreeFromCalls } from "./utils/tree.js";
+import { mockClient } from "./utils/mock_client.js";
+import { FakeChatModel } from "@langchain/core/utils/testing";
+import { ChatPromptTemplate } from "@langchain/core/prompts";
+import { StringOutputParser } from "@langchain/core/output_parsers";
+import { LangChainTracer } from "@langchain/core/tracers/tracer_langchain";
+import { BaseMessage, HumanMessage } from "@langchain/core/messages";
+import { awaitAllCallbacks } from "@langchain/core/callbacks/promises";
+import { RunnableTraceable, getLangchainCallbacks } from "../langchain.js";
+
+describe("to langchain", () => {
+  const llm = new FakeChatModel({});
+  const prompt = ChatPromptTemplate.fromMessages<{ text: string }>([
+    ["human", "{text}"],
+  ]);
+  const parser = new StringOutputParser();
+  const chain = prompt.pipe(llm).pipe(parser);
+
+  test("invoke", async () => {
+    const { client, callSpy } = mockClient();
+
+    const main = traceable(
+      async (input: { text: string }) => {
+        return chain.invoke(input, {
+          callbacks: await getLangchainCallbacks(),
```

Comment:
Apologies if I'm still not getting it - there's no way to do this automatically?

I thought part of the idea was to pull in the local storage instance from `@langchain/core/singletons` so that this would happen seamlessly?

Reply:
Resolved in Slack